src/main.rs

```diff
@@ -0,0 +1,77 @@
+mod dockerfile;
+
+use dockerfile::{Dockerfile, ValidationResult};
+use glob::glob;
+use std::{error::Error, process};
+
+fn main() {
+    let dockerfiles = get_dockerfiles().expect("[ERROR] Failed to get Dockerfiles.");
+    if dockerfiles.is_empty() {
+        skip();
+    };
+
+    let invalid = validate_dockerfiles(&dockerfiles);
+    if invalid.is_empty() {
+        pass();
+    }
+
+    fail(&invalid);
+}
+
+fn get_dockerfiles() -> Result<Vec<Dockerfile>, Box<dyn Error>> {
+    glob("**/*Dockerfile*")?
+        .map(|path| Dockerfile::new(path?))
+        .collect()
+}
+
+fn validate_dockerfiles(dockerfiles: &[Dockerfile]) -> Vec<(&Dockerfile, ValidationResult)> {
```

Comment:
Should I be returning `Vec<(Dockerfile, ValidationResult)>` here instead?

Reply:
*gestures at the other comment of it depending*

Reply:
I wouldn't. Your ownership is maintained intact as it's never owned by this function ;)